### PR TITLE
Fix run_integration_tests.sh compatibility on MacOS

### LINKED ISSUE #388: GoogleHadoopFileSystemIntegrationHadoop2Test fails in V1.9.17
I am trying to use this feature. https://github.com/GoogleCloudDataproc/hadoop-connectors/commit/02b17d1b11369e53a75f48f2ce9e064e040070d9

This contains GoogleHadoopFileSystemIntegrationHadoop2Test , so I tried running it but it fails saying account not valid even though I am able to use my account for listing and updates. 

I can see that this test has been removed from latest version. Could somebody help me running this test as well as make me understand why this class has been removed. Also if this class has been removed, where are the tests for feature "Add Hadoop File System extended attributes support."

Thanks.

### PR DESCRIPTION
**Pull request by medb**
Fixes #388

### CODE REVIEW

File: tools/run_integration_tests.sh [hunk 1]
```diff
@@ -56,7 +56,12 @@ check_required_params
 
 # When tests run, they run in the root of the module directory. Anything
 # relative to our current directory won't work properly
-GCS_TEST_PRIVATE_KEYFILE=$(readlink -f "${GCS_TEST_PRIVATE_KEYFILE}")
+if [[ $(uname -s) == Darwin ]]; then
+  # On MacOS `readlink` does not support `-e` parameter
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "${GCS_TEST_PRIVATE_KEYFILE}")
```
Comment (mukund-thakur) [inline tools/run_integration_tests.sh:61]: What would the behaviour here if an absolute credential file path is passed?

Comment (medb) [reply tools/run_integration_tests.sh]: Fixed it and added assert for keyfile existence after absolute path and symlink resolution.

Comment (mukund-thakur) [reply tools/run_integration_tests.sh]: Great Thanks :)

File: tools/run_integration_tests.sh [hunk 2]
```diff
@@ -50,14 +50,30 @@ check_required_params() {
...
 # When tests run, they run in the root of the module directory. Anything
 # relative to our current directory won't work properly
-GCS_TEST_PRIVATE_KEYFILE=$(readlink -f "${GCS_TEST_PRIVATE_KEYFILE}")
+if [[ $(uname -s) == Darwin ]]; then
+  # On MacOS `readlink` does not support `-f` parameter
+  # so we need manually resolve absolute path and symlink after that
+  abs_path() {    
+   echo "$(cd $(dirname "$1"); pwd)/$(basename "$1")"
+  }
+  GCS_TEST_PRIVATE_KEYFILE=$(abs_path "$GCS_TEST_PRIVATE_KEYFILE")
```
Comment (cyxxy) [inline tools/run_integration_tests.sh:63]: be consistent and use curly brackets ${GCS_TEST_PRIVATE_KEYFILE}

File: tools/run_integration_tests.sh [hunk 3]
```diff
@@ -50,14 +50,30 @@ check_required_params() {
...
+  while [[ -L $GCS_TEST_PRIVATE_KEYFILE ]]; do
+    GCS_TEST_PRIVATE_KEYFILE=$(readlink -n "$GCS_TEST_PRIVATE_KEYFILE")
```
Comment (cyxxy) [inline tools/run_integration_tests.sh:65]: ditto

File: tools/run_integration_tests.sh [hunk 4]
```diff
@@ -50,14 +50,30 @@ check_required_params() {
...
+else
+  GCS_TEST_PRIVATE_KEYFILE=$(readlink -n -f "${GCS_TEST_PRIVATE_KEYFILE}")
+fi
+if [[ ! -f $GCS_TEST_PRIVATE_KEYFILE ]]; then
+  echo "Resolved '${GCS_TEST_PRIVATE_KEYFILE}' keyfile does not exist."
+  exit 1
+fi
+
 export GCS_TEST_PRIVATE_KEYFILE
+export HDFS_ROOT=file:///tmp
```
Comment (cyxxy) [inline tools/run_integration_tests.sh:76]: why is this?

File: tools/run_integration_tests.sh (hunk 2 above)
Comment (medb) [reply tools/run_integration_tests.sh]: Done.

File: tools/run_integration_tests.sh (hunk 4 above)
Comment (medb) [reply tools/run_integration_tests.sh:76]: Done.

File: tools/run_integration_tests.sh (hunk 3 above)
Comment (medb) [reply tools/run_integration_tests.sh]: Done.

**Pull request merged**